jobs/dtd-metrics/src/index.ts

```diff
@@ -12,81 +11,60 @@ import {
   // getTenantDistributionMetric,
   // getTenantSignupsTrendMetric,
   // getOnboardedTenantsCountByMacroCategoriesMetric,
-} from './services/index.js'
-import { GithubClient } from './services/github-client.service.js'
-import { wrapPromiseWithLogs } from './utils/helpers.utils.js'
-import { GlobalStoreService } from './services/global-store.service.js'
+} from './metrics/index.js'
+import { GithubClient, GlobalStoreService } from './services/index.js'
+import { MetricsProducerService } from './services/metrics-producer.service.js'
+import { log } from './utils/helpers.utils.js'
 
-const log = console.log
+log.info('Starting program\n')
 
-let readModel: ReadModelClient
-
-async function main(): Promise<void> {
-  log('Starting program\n')
-
-  readModel = await ReadModelClient.connect({
-    mongodbReplicaSet: env.MONGODB_REPLICA_SET,
-    mongodbDirectConnection: env.MONGODB_DIRECT_CONNECTION,
-    mongodbReadPreference: env.MONGODB_READ_PREFERENCE as ReadPreferenceMode,
-    mongodbRetryWrites: env.MONGODB_RETRY_WRITES,
-    readModelDbHost: env.READ_MODEL_DB_HOST,
-    readModelDbPort: env.READ_MODEL_DB_PORT,
-    readModelDbUser: env.READ_MODEL_DB_USER,
-    readModelDbPassword: env.READ_MODEL_DB_PASSWORD,
-    readModelDbName: env.READ_MODEL_DB_NAME,
-  })
+const readModel = await ReadModelClient.connect({
+  mongodbReplicaSet: env.MONGODB_REPLICA_SET,
+  mongodbDirectConnection: env.MONGODB_DIRECT_CONNECTION,
+  mongodbReadPreference: env.MONGODB_READ_PREFERENCE as ReadPreferenceMode,
+  mongodbRetryWrites: env.MONGODB_RETRY_WRITES,
+  readModelDbHost: env.READ_MODEL_DB_HOST,
+  readModelDbPort: env.READ_MODEL_DB_PORT,
+  readModelDbUser: env.READ_MODEL_DB_USER,
+  readModelDbPassword: env.READ_MODEL_DB_PASSWORD,
+  readModelDbName: env.READ_MODEL_DB_NAME,
+})
 
+try {
   const githubClient = new GithubClient(env.GITHUB_ACCESS_TOKEN)
   const awsS3BucketClient = new AwsS3BucketClient(env.STORAGE_BUCKET)
 
-  log('Retrieving metrics...')
+  log.info('Initializing global store...')
+  const globalStore = await GlobalStoreService.init(readModel, { cache: env.CACHE_GLOBAL_STORE })
+  log.info('Global store initialized!\n')
```

Comment:
new lines in logs create multiple entries on cloudwatch with empty data, and should be avoided